scripts/reliability_metrics.py

```diff
@@ -270,37 +72,71 @@ def init_states():
         StateFactory.register_state('FAULT', FaultState)
 
 
+class Utils:
+    args: argparse.Namespace = None
+    handler: logging.Handler = None
+    loggers: dict = {}
+    logger_adapters: dict = {}
+
+    @staticmethod
+    def get_arguments() -> argparse.Namespace:
+        if Utils.args is not None:
+            return Utils.args
+        parser: argparse.ArgumentParser = argparse.ArgumentParser(description='find services errors in syslog')
+        parser.add_argument('state', choices=['MASTER', 'BACKUP', 'FAULT'], help='state of the machine')
+        Utils.args = parser.parse_args()
+        return Utils.args
 
-def get_arguments() -> argparse.Namespace:
-    parser: argparse.ArgumentParser = argparse.ArgumentParser(description='find services errors in syslog')
-    parser.add_argument('state', choices=['MASTER', 'BACKUP', 'FAULT'], help='state of the machine')
-    return parser.parse_args()
+    @staticmethod
+    def get_file_handler() -> logging.Handler:
+        if Utils.handler is not None:
+            return Utils.handler
 
+        file_name: str = f'/tmp/{os.path.basename(__file__).split(".")[0]}.log'
```

Comment:
1. better to use : `os.path.join` or `os.sep`
2. please use `os.path.splitext`

Reply:
https://github.com/ottopia-tech/keepalived-exporter/pull/5/commits/5b556e9425408ff5916e0b979b3f48f721a06de3